packages/rest/src/router/openapi-path.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright IBM Corp. 2017, 2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import pathToRegExp = require('path-to-regexp');
+
+/**
+ * OpenAPI spec 3.x does not specify the valid forms of path templates.
+ *
+ * Other ones such as [URI Template](https://tools.ietf.org/html/rfc6570#section-2.3)
+ * or [path-to-regexp](https://github.com/pillarjs/path-to-regexp#named-parameters)
+ * allows `[A-Za-z0-9_]`
+ */
+const POSSIBLE_VARNAME_PATTERN = /\{([^\}]+)\}/g;
+const INVALID_VARNAME_PATTERN = /\{([^\}]*[^\w\}][^\}]*)\}/;
+
+/**
+ * Validate the path to be compatible with OpenAPI path template. No parameter
+ * modifier, custom pattern, or unnamed parameter is allowed.
+ */
+export function validatePath(path: string = '/') {
```

Comment:
Perhaps `validateApiPath` would be a better name, to make it clear that we are validating REST API/URL paths and not filesystem paths?